apps/opik-backend/src/main/java/com/comet/opik/domain/DatasetItemDAO.java

```diff
@@ -794,10 +808,30 @@ public Mono<DatasetItemPage> getItems(@NonNull UUID datasetId, int page, int siz
                         })));
     }
 
-    private Publisher<Map.Entry<Long, Set<Column>>> mapCount(Result result) {
+    private Map.Entry<Long, Set<Column>> groupResults(Map.Entry<Long, Set<Column>> result1,
+            Map.Entry<Long, Set<Column>> result2) {
+        return Map.entry(result1.getKey() + result2.getKey(),
+                Sets.union(result1.getValue(), result2.getValue()));
+    }
+
+    private Publisher<Map.Entry<Long, Set<Column>>> mapCountAndColumns(Result result) {
         return result.map((row, rowMetadata) -> Map.entry(
-                row.get(0, Long.class),
-                ((Map<String, String[]>) Optional.ofNullable(row.get(1, Map.class)).orElse(Map.of()))
+                row.get("count", Long.class),
+                ((Map<String, String[]>) Optional.ofNullable(row.get("columns", Map.class)).orElse(Map.of()))
+                        .entrySet()
+                        .stream()
+                        .map(columnArray -> new Column(columnArray.getKey(),
+                                Set.of(mapColumnType(columnArray.getValue()))))
+                        .collect(Collectors.toSet())));
+    }
+
+    private Publisher<Long> mapCount(Result result) {
+        return result.map((row, rowMetadata) -> row.get(0, Long.class));
+    }
+
+    private Mono<Set<Column>> mapColumns(Result result) {
+        return Mono.from(result.map((row,
+                rowMetadata) -> ((Map<String, String[]>) Optional.ofNullable(row.get(0, Map.class)).orElse(Map.of()))
```

Comment:
Minor: also, better use the name (`columns`) instead of the position, like you do above.